test/blackbox-tests/test-cases/file-available.t

```diff
@@ -0,0 +1,77 @@
+Test action %{file-available:<path>} where '<path>' is any path to a file 
+within the workspace and outside of the build directory.
+
+Populate the current workspace with some files and test if they are available.
+
+  $ mkdir bin
+  $ touch bin/foo.ml
+  $ touch bar.ml
+
+  $ cat > dune-project <<EOF
+  > (lang dune 3.0)
+  > EOF
+
+  $ cat > dune <<EOF
+  > (rule
+  >   (alias runtest0)
+  >   (action 
+  >     (progn
+  >       (echo "dune-project available: %{file-available:dune-project}\n")
+  >       (echo "bin/foo.ml available: %{file-available:bin/foo.ml}\n")
+  >       (echo "bar.ml available: %{file-available:bar.ml}\n"))))
+  > EOF
+
+  $ dune build @runtest0
+  dune-project available: true
+  bin/foo.ml available: true
+  bar.ml available: true
+
+Test to see that some files that do not exist are not available.
+
+  $ cat > dune <<EOF
+  > (rule
+  >   (alias runtest1)
+  >   (action
+  >     (progn
+  >       (echo "baz.ml available: %{file-available:baz.ml}\n")
+  >       (echo "bin/test.ml available: %{file-available:bin/test.ml}\n"))))
+  > EOF
+
+  $ dune build @runtest1
+  baz.ml available: false
+  bin/test.ml available: false
+
+Test that files generated by rules are recognized as available.
+
+  $ cat > dune <<EOF
+  > (rule
+  >   (with-stdout-to gen.txt (echo "generated file text")))
+  > (rule
+  >   (write-file gen1.txt "another generated file")) 
+  > (rule
+  >   (alias runtest2)
+  >   (action
+  >     (progn
+  >       (echo "gen.txt available: %{file-available:_build/default/gen.txt}\n")
+  >       (echo "gen1.txt available: %{file-available:_build/default/gen1.txt}\n"))))
```

Comment:
You should not have to qualify the context with _build/default. The rules must do that for you.

Reply:
Could you expand upon this a bit, I'm not sure I fully understand what you mean. How I interpret this is, by default the macro invocation should be looking for files staring in the base directory `_build/default`?

Reply:
All rules have the following property: they work in any "context". The path `_build/default/gen.txt` that you referred is a path from the `default` context. So it means that your rule cannot possibly work for other contexts. This is because you have implemented `file-available` to only work for source files.